interfaces/acados_c/ocp_qp.h

```diff
@@ -0,0 +1,91 @@
+/*
+ *    This file is part of acados.
+ *
+ *    acados is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation; either
+ *    version 3 of the License, or (at your option) any later version.
+ *
+ *    acados is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with acados; if not, write to the Free Software Foundation,
+ *    Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#ifndef ACADOS_C_OCP_QP_H_
+#define ACADOS_C_OCP_QP_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// acados
+#include <acados/ocp_qp/ocp_qp_common.h>
+#include <acados/ocp_qp/ocp_qp_common_frontend.h>
+// acados_c
+#include "acados_c/common.h"
+
+typedef enum {
+    PARTIAL_CONDENSING_HPIPM,
```

Comment:
I thought ultimately it would be possible to also pass HPIPM here that would create an ocp_qp solver without partial condensing (i.e. without the nesting the sparse solver in the sparse_solver module). That is my last and only "major" remark of the PR.

Reply:
so I would additionally have entries HPIPM, HPMPC, QPDUNES, OOQP here and a proper handling in ocp_qp.c

Reply:
Indeed, the implementation is on my planning.